test/provider_os_memory.cpp

```diff
@@ -147,6 +147,34 @@ TEST_F(test, provider_os_memory_alloc_free_2k_alignment_2k) {
     umfMemoryProviderDestroy(os_memory_provider);
 }
 
+TEST_F(test, provider_os_memory_alloc_free_4100_alignment_16k) {
+    umf_result_t umf_result;
+    umf_memory_provider_handle_t os_memory_provider = nullptr;
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+
+    umf_result = umfMemoryProviderCreate(&UMF_OS_MEMORY_PROVIDER_OPS,
+                                         &os_memory_provider_params,
+                                         &os_memory_provider);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(os_memory_provider, nullptr);
+
+    void *ptr = nullptr;
+    size_t size = SIZE_4K + 4;
+    size_t alignment = 4 * SIZE_4K;
+    umf_result =
+        umfMemoryProviderAlloc(os_memory_provider, size, alignment, &ptr);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
```

Comment:
please check here if the ptr is aligned correctly

Reply:
Done